open-ce/validate_config.py

```diff
@@ -26,52 +26,58 @@ def make_parser():
                                                variant['mpi_type'],
-                                               repository_folder=args.repository_folder,
-                                               conda_build_config=args.conda_build_config)
+                                               repository_folder=repository_folder,
+                                               conda_build_config=conda_build_config)
+                validate_build_tree(recipes, variant)
             except OpenCEError as err:
-                raise OpenCEError(Error.VALIDATE_CONFIG, args.conda_build_config, env_file, variant, err.msg) from err
+                raise OpenCEError(Error.VALIDATE_CONFIG, conda_build_config, env_file, variant, err.msg) from err
+            print('Successfully validated {} for {} : {}'.format(conda_build_config, env_file, variant))
 
-            packages = [package for recipe in recipes for package in recipe.packages]
-            channels = {channel for recipe in recipes for channel in recipe.channels}
-            deps = {dep for recipe in recipes for dep in recipe.run_dependencies}
-            deps.update(recipes.get_external_dependencies(variant))
+        print('Successfully validated {} for {}'.format(conda_build_config, variant))
 
-            pkg_args = " ".join(["\"{}\"".format(utils.generalize_version(dep)) for dep in deps
-                                                                          if not utils.remove_version(dep) in packages])
+    print("{} Successfully validated!".format(conda_build_config))
 
-            channel_args = " ".join({"-c \"{}\"".format(channel) for channel in channels})
-
-            cli = "conda create --dry-run -n test_conda_dependencies {} {}".format(channel_args, pkg_args)
-
-            retval = utils.run_and_log(cli)
+def validate_build_tree(recipes, variant):
+    '''
+    Check a build tree for dependency compatability.
+    '''
+    packages = [package for recipe in recipes for package in recipe.packages]
+    channels = {channel for recipe in recipes for channel in recipe.channels}
+    deps = {dep for recipe in recipes for dep in recipe.run_dependencies}
+    deps.update(recipes.get_external_dependencies(variant))
 
-            if retval != 0:
-                raise OpenCEError(Error.VALIDATE_CONFIG, args.conda_build_config, env_file, variant, "")
+    pkg_args = " ".join(["\"{}\"".format(utils.generalize_version(dep)) for dep in deps
+                                                                    if not utils.remove_version(dep) in packages])
 
-            print('Successfully validated {} for {} : {}'.format(args.conda_build_config, env_file, variant))
+    channel_args = " ".join({"-c \"{}\"".format(channel) for channel in channels})
 
-        print('Successfully validated {} for {}'.format(args.conda_build_config, variant))
+    cli = "conda create --dry-run -n test_conda_dependencies {} {}".format(channel_args, pkg_args)
```

Comment:
Maybe we should capture the output from this and only display it if there is an error.

Reply:
All of this output is displayed when it is executed.
I could capture it instead and include it as part of the exception.

Reply:
This is done.